airbyte-server/src/test/java/io/airbyte/server/handlers/SourceDefinitionsHandler2Test.java

```diff
@@ -0,0 +1,293 @@
+        SynchronousJobMetadata.mock(ConfigType.GET_SPEC)));
+
+    final ActorDefinition updatedSource = Jsons.clone(this.sourceDefinition).withDockerImageTag(newDockerImageTag).withSpec(newSpec);
+
+    final SourceDefinitionRead sourceDefinitionRead = sourceDefinitionsHandler
+        .updateActorDefinition(new SourceDefinitionUpdate().sourceDefinitionId(this.sourceDefinition.getId()).dockerImageTag(newDockerImageTag));
+
+    assertEquals(newDockerImageTag, sourceDefinitionRead.getDockerImageTag());
+    verify(schedulerSynchronousClient).createGetSpecJob(newImageName);
+    verify(configRepository).writeActorDefinition(updatedSource);
+  }
+
+  @Test
+  @DisplayName("deleteSourceDefinition should correctly delete a sourceDefinition")
+  void testDeleteSourceDefinition() throws ConfigNotFoundException, IOException, JsonValidationException {
+    final SourceDefinitionIdRequestBody sourceDefinitionIdRequestBody =
+        new SourceDefinitionIdRequestBody().sourceDefinitionId(sourceDefinition.getId());
+    final ActorDefinition updatedSourceDefinition = Jsons.clone(this.sourceDefinition).withTombstone(true);
+    final SourceRead source = new SourceRead();
+
+    when(configRepository.getActorDefinition(sourceDefinition.getId(), ActorType.SOURCE))
+        .thenReturn(sourceDefinition);
+    when(sourceHandler.listSourcesForSourceDefinition(sourceDefinitionIdRequestBody))
+        .thenReturn(new SourceReadList().sources(Collections.singletonList(source)));
+
+    assertFalse(sourceDefinition.getTombstone());
+
+    sourceDefinitionsHandler.deleteActorDefinition(sourceDefinitionIdRequestBody);
+
+    verify(sourceHandler).deleteSource(source);
+    verify(configRepository).writeActorDefinition(updatedSourceDefinition);
+  }
+
+  @Nested
+  @DisplayName("listLatest")
+  class listLatest {
+
+    @Test
+    @DisplayName("should return the latest list")
+    void testCorrect() throws InterruptedException {
+      final ActorDefinition sourceDefinition = generateSourceDefinition();
+      when(githubStore.getLatestDefinitions(ActorType.SOURCE)).thenReturn(Collections.singletonList(sourceDefinition));
+
+      final var sourceDefinitionReadList = sourceDefinitionsHandler.listLatestActorDefinitions().getSourceDefinitions();
+      assertEquals(1, sourceDefinitionReadList.size());
+
+      final var sourceDefinitionRead = sourceDefinitionReadList.get(0);
+      assertEquals(SourceDefinitionsHandler.buildSourceDefinitionRead(sourceDefinition), sourceDefinitionRead);
+    }
+
+    @Test
+    @DisplayName("returns empty collection if cannot find latest definitions")
+    void testHttpTimeout() {
+      assertEquals(0, sourceDefinitionsHandler.listLatestActorDefinitions().getSourceDefinitions().size());
+    }
+
+    @Test
+    @DisplayName("Icon should contain data")
+    void testIconHoldsData() {
+      final String icon = SourceDefinitionsHandler.loadIcon(sourceDefinition.getIcon());
+      assertNotNull(icon);
+      assert (icon.length() > 3000);
```

Comment:
Consider using `assertTrue` instead of the hard `assert` keyword.  This would allow us to take advantage of JUnit's value comparison in the output explaining what is not true here, as opposed to the hard failure of the JVM via the `assert` keyword.

Reply:
agreed. this is copy pasta. i copied the original tests so that i could verify that my migrations didn't cause regressions. but they could definitely use some love.